### PR TITLE
Unify Auth and Contract invoke logic

### PR DESCRIPTION
**Pull request by Kolezhniuk**

### CODE REVIEW

File: src/iden3comm/handlers/common.ts [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+ * @returns A promise that resolves to an array of zero-knowledge proof responses.
+ */
+export const processProtocolRequests = async (
+  senderIdentifier: DID,
+  request: BasicMessage,
+  proofService: IProofService,
+  opts: {
+    mediaType?: MediaType;
+    packerOptions?: JWSPackerParams;
+    allowedCircuits: CircuitId[];
```
Comment (volodymyr-basiuk) [inline src/iden3comm/handlers/common.ts:77]: maybe `supportedCircuits` ?

File: src/iden3comm/handlers/common.ts [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+ * Processes protocol requests and generates zero-knowledge proof responses.
+ *
+ * @param senderIdentifier - The DID identifier.
+ * @param request - protocol message.
+ * @param opts - additional options.  verifierDid is the DID of the verifier.
+ * @param opts.allowedCircuits - allowedCircuits is a list of allowed circuits.
+ * @param opts.verifierDid - verifierDid is the DID of the verifier.
+ * @returns A promise that resolves to an array of zero-knowledge proof responses.
+ */
+export const processProtocolRequests = async (
```
Comment (vmidyllic) [inline src/iden3comm/handlers/common.ts:70]: rename processZeroKnowledgeProofRequests

File: tests/handlers/contract-request.test.ts [hunk 3]
```diff
@@ -446,7 +450,7 @@ describe('contract-request', () => {
     );
   });
   // V3 integration test
-  it.skip('contract request flow V3 - integration test', async () => {
+  it.only('contract request flow V3 - integration test', async () => {
```
Comment (volodymyr-basiuk) [inline tests/handlers/contract-request.test.ts:453]: rm only

File: src/iden3comm/handlers/common.ts [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+ * @param request - protocol message.
+ * @param opts - additional options.  verifierDid is the DID of the verifier.
+ * @param opts.allowedCircuits - allowedCircuits is a list of allowed circuits.
+ * @param opts.verifierDid - verifierDid is the DID of the verifier.
+ * @returns A promise that resolves to an array of zero-knowledge proof responses.
+ */
+export const processZeroKnowledgeProofRequests = async (
+  senderIdentifier: DID,
+  requests: ZeroKnowledgeProofRequest[] | undefined,
+  from: DID | undefined,
```
Comment (volodymyr-basiuk) [inline src/iden3comm/handlers/common.ts:72]: add from and proofService to @params

**Review by volodymyr-basiuk — APPROVED** [review]

**Review by vmidyllic — APPROVED** [review]

**Pull request merged**